src/ComputeSharp/Graphics/Extensions/ReadBackTexture2DExtensions.cs

```diff
@@ -8,6 +9,52 @@ namespace ComputeSharp;
+    /// <typeparam name="T">The type of items stored on the texture.</typeparam>
+    /// <param name="destination">The target <see cref="ReadBackTexture2D{T}"/> instance to write data to.</param>
+    /// <param name="source">The input <see cref="Texture2D{T}"/> instance to read data from.</param>
+    public static void CopyFrom<T>(this ReadBackTexture2D<T> destination, Texture2D<T> source)
+        where T : unmanaged
+    {
+        source.CopyTo(destination, 0, 0, 0, 0, source.Width, source.Height);
+    }
+
+    /// <summary>
+    /// Reads the contents of a <see cref="Texture2D{T}"/> instance and writes them into a target <see cref="ReadBackTexture2D{T}"/> instance.
+    /// </summary>
+    /// <typeparam name="T">The type of items stored on the texture.</typeparam>
+    /// <param name="destination">The target <see cref="ReadBackTexture2D{T}"/> instance to write data to.</param>
+    /// <param name="source">The input <see cref="Texture2D{T}"/> instance to read data from.</param>
+    /// <param name="sourceOffsetX">The horizontal offset in the source texture.</param>
+    /// <param name="sourceOffsetY">The vertical offset in the source texture.</param>
+    /// <param name="width">The width of the memory area to copy.</param>
+    /// <param name="height">The height of the memory area to copy.</param>
+    public static void CopyFrom<T>(this ReadBackTexture2D<T> destination, Texture2D<T> source, int sourceOffsetX, int sourceOffsetY, int width, int height)
+        where T : unmanaged
+    {
+        source.CopyTo(destination, sourceOffsetX, sourceOffsetY, 0, 0, width, height);
+    }
+
+    /// <summary>
+    /// Reads the contents of a <see cref="Texture2D{T}"/> instance and writes them into a target <see cref="ReadBackTexture2D{T}"/> instance.
+    /// </summary>
+    /// <typeparam name="T">The type of items stored on the texture.</typeparam>
+    /// <param name="destination">The target <see cref="ReadBackTexture2D{T}"/> instance to write data to.</param>
+    /// <param name="source">The input <see cref="Texture2D{T}"/> instance to read data from.</param>
+    /// <param name="sourceOffsetX">The horizontal offset in the source texture.</param>
+    /// <param name="sourceOffsetY">The vertical offset in the source texture.</param>
+    /// <param name="destinationOffsetX">The horizontal offset in the destination texture.</param>
+    /// <param name="destinationOffsetY">The vertical offset in the destination texture.</param>
+    /// <param name="width">The width of the memory area to copy.</param>
+    /// <param name="height">The height of the memory area to copy.</param>
+    public static void CopyFrom<T>(this ReadBackTexture2D<T> destination, Texture2D<T> source, int sourceOffsetX, int sourceOffsetY, int destinationOffsetX, int destinationOffsetY, int width, int height)
+        where T : unmanaged
+    {
+        source.CopyTo(destination, sourceOffsetX, sourceOffsetY, destinationOffsetX, destinationOffsetY, width, height);
```

Comment:
Assuming the base function does all the bounds checking?

Reply:
That is correct, this calls `Texture2D<T>.CopyTo`, which does all the paarameter validation:

https://github.com/Sergio0694/ComputeSharp/blob/da90dd9157aaa07ca6834a50380e65ce1914a0f3/src/ComputeSharp/Graphics/Resources/Abstract/Texture2D%7BT%7D.cs#L313-L324